packages/dx-grid-core/src/plugins/integrated-grouping/computeds.ts

```diff
@@ -29,11 +29,13 @@ export const groupedRows: GroupedRowsFn = (
     || defaultColumnCriteria;
   const groups = rows
     .reduce((acc, row) => {
-      const { key, value } = groupCriteria(getCellValue(row, columnName), row);
+      const rawValue = getCellValue(row, columnName);
+      const { key, value } = groupCriteria(rawValue, row);
       const sameKeyItems = acc.get(key);
 
       if (!sameKeyItems) {
-        acc.set(key, [value || key, key, [row]]);
+        const groupingValue = value === rawValue ? value : value || key;
```

Comment:
Just a question: is there any case when value !== rawValue? I've tried to use `undefined`, `null` as a column value. `value === rawValue` - is always `true`.

Reply:
Yes - when custom columnCriteria is specified and it returns only a key, thus a value is undefined but rawValue may be not